consensus/service/src/config/mod.rs

```diff
@@ -26,96 +28,83 @@ pub struct Config {
-    #[structopt(long = "network", parse(from_os_str))]
+    #[clap(long = "network", env, parse(from_os_str))]
     pub network_path: PathBuf,
 
     /// Your Intel IAS API key.
-    #[structopt(long)]
+    #[clap(long, env)]
     pub ias_api_key: String,
 
     /// The Service Provider ID (SPID) associated with your Intel IAS API Key.
-    #[structopt(long)]
+    #[clap(long, env)]
     pub ias_spid: ProviderId,
 
     /// The location on which to listen for peer traffic.
     ///
     /// The local node id is derived from the peer_listen_uri.
-    #[structopt(long, default_value = "insecure-mcp://0.0.0.0:8080/")]
+    #[clap(long, env, default_value = "insecure-mcp://0.0.0.0:8080/")]
     pub peer_listen_uri: PeerUri,
 
     /// Client listening URI.
-    #[structopt(long, default_value = "insecure-mc://0.0.0.0:3223/")]
+    #[clap(long, env, default_value = "insecure-mc://0.0.0.0:3223/")]
     pub client_listen_uri: ClientUri,
 
     /// Optional admin listening URI.
-    #[structopt(long)]
+    #[clap(long, env)]
     pub admin_listen_uri: Option<AdminUri>,
 
     /// The location to write the externalized blocks for the ledger.
-    #[structopt(long, parse(from_os_str))]
+    #[clap(long, env, parse(from_os_str))]
     pub ledger_path: PathBuf,
 
     /// The location from which to load the origin block.
-    #[structopt(long, parse(from_os_str))]
+    #[clap(long, env, parse(from_os_str))]
     pub origin_block_path: Option<PathBuf>,
 
     /// SCP debug output.
-    #[structopt(long, parse(from_os_str))]
+    #[clap(long, env, parse(from_os_str))]
     pub scp_debug_dump: Option<PathBuf>,
 
     /// Path to the sealed block signing key
-    #[structopt(long, parse(from_os_str))]
+    #[clap(long, env, parse(from_os_str))]
     pub sealed_block_signing_key: PathBuf,
 
     /// Enables authenticating client requests using Authorization tokens using
     /// the provided hex-encoded 32 bytes shared secret.
-    #[structopt(long, parse(try_from_str=hex::FromHex::from_hex))]
+    #[clap(long, env, parse(try_from_str = hex::FromHex::from_hex))]
     pub client_auth_token_secret: Option<[u8; 32]>,
 
     /// Maximal client authentication token lifetime, in seconds (only relevant
     /// when --client-auth-token-secret is used. Defaults to 86400 - 24
     /// hours).
-    #[structopt(long, default_value = "86400", parse(try_from_str=parse_duration_in_seconds))]
+    #[clap(long, env, default_value = "86400", parse(try_from_str = parse_duration_in_seconds))]
     pub client_auth_token_max_lifetime: Duration,
 
     /// The location for the network.toml/json configuration file.
-    #[structopt(long = "tokens", parse(from_os_str))]
+    #[clap(long = "tokens", env, parse(from_os_str))]
     pub tokens_path: Option<PathBuf>,
 
     /// The configured block version
-    #[structopt(long, env = "MC_BLOCK_VERSION", default_value = "1")]
+    #[clap(long, env, default_value = "1", parse(try_from_str = parse_block_version))]
```

Comment:
This changes the env variable from `MC_BLOCK_VERSION` to `BLOCK_VERSION`

FYI @jgreat

Reply:
Could we leave this as-is?

Reply:
Could we flip this and make everything that's newly `env`'d `MC_`?

Reply:
`clap` does not support this, so we'd have to spell out the env var for each flag. I filed https://github.com/clap-rs/clap/issues/3517 requesting support for a prefix for env vars.

Reply:
Sure, but I'm OK defining them as MC_FOO everywhere, just because they don't exist now, and it will be painful to change it later.

Reply:
Switched to manually specifying MC_FOO everywhere.